test/integration/connect/envoy/case-cfg-resolver-cluster-peering-failover/primary/verify.bats

```diff
@@ -85,3 +91,27 @@ load helpers
 @test "s1 upstream made 1 connection to s2 through the cluster peer" {
   assert_envoy_metric_at_least 127.0.0.1:19000 "cluster.failover-target~s2.default.primary-to-alpha.external.*cx_total" 1
 }
+
+# Redirect
+
+@test "reset envoy statistics for redirect" {
+  reset_envoy_metrics 127.0.0.1:19000
+}
+
+@test "s1 upstream should have healthy endpoints for virtual-s2" {
+  assert_upstream_has_endpoints_in_status 127.0.0.1:19000 s2.default.primary-to-alpha.external HEALTHY 1
+}
+
+@test "s1 upstream should have healthy endpoints for s2 in the cluster peer" {
+  assert_upstream_has_endpoints_in_status 127.0.0.1:19000 s2.default.primary-to-alpha.external HEALTHY 1
+}
```

Comment:
These are the same steps. I think you wanted to test the virtual service in the first one, correct?

Reply:
I remember how this works now. The cluster name is the name of the destination cluster and doesn't take the local service name into account. For example, in our tests we [redirect db to db in cluster peer cluster-01](https://github.com/hashicorp/consul/pull/14445/files#diff-55a80e75e8a5a8ed1d8360d217c766ebbe0949abf67ce3d51195ac6e2d8fe9c2R311-R320). The resulting cluster is [named `db.default.cluster-01.external.peer1.domain` rather than the local db cluster name](https://github.com/hashicorp/consul/pull/14445/files#diff-e61d0ee1542c3bcd0de50f6fb9eff56f35bc209cfa013e0c162c3ef18e2aff58R6).

I removed the duplicate assertion here.